### DISCUSSION TITLE
Help on configuring SSL client certificate stored in Windows local folder

### DISCUSSION THREAD

**Discussion by mith-emsa** [thunderclient/thunder-client-support#876] [Q&A] [answered]
I tried to follow the instructions on https://github.com/rangav/thunder-client-support#auth and added the path to the PFX file that I have locally stored in a folder using the option mentioned in the instructions
```
"thunder-client.certificates": [
        {
            "host": "localhost:8081",
            "pfxPath": "C:/Users/terrami/Documents/HP-IMS Related Documents/SSL Certificates/MyCert.pfx",
            "passphrase": "xxxx"
        },
    ]

```
I also tried another option which was to use 
```
"thunder-client.certificates": [
        {
            "host": "localhost:8081",
            "pfxPath": "C:\\Users\\terrami\\Documents\\HP-IMS Related Documents\\SSL Certificates\\MyCert.pfx",
            "passphrase": "xxxx"
        },
    ]

```
With both options when I send an HTTP GET request the client certificate is not used by Thunder Client. I also did not see any option to select the certificate. Could also not find any further instructions on this. Is there any restriction on where to place the certificate? My port 8081 on localhost is not available. But I do not understand why that is needed for a file that is stored locally. Should that be removed? 
Any help on this would be greatly appreciated. 
Many thanks

**Comment by rangav** [comment]
Hi @mith-emsa thanks fore reporting, will verify and get back to you

**Reply by mith-emsa** [reply]
Dear @rangav  are there any news on my question? Thanks

**Reply by rangav** [reply]
can you try this?

`C:\Users\terrami\Documents\HP-IMS Related Documents\SSL Certificates\MyCert.pfx`

**Reply by mith-emsa** [reply]
Dear @rangav  I have tried and got the same error that the client certificate is missing. Does not seem to be an option anywhere to select to send the client certificate with the HTTP request.

**Reply by rangav** [reply]
can you see any logs in Output window (select TC)

**Reply by mith-emsa** [reply]
I only see the response which is
{
  "statusCode": 401,
  "message": "Client certificate missing"
}

Do not see any window to check for logs.

**Reply by rangav** [reply]
you can see logs here
![image](https://github.com/rangav/thunder-client-support/assets/8637550/60f135ed-87f2-4e92-a3bc-27ca05f3278f)

**Reply by mith-emsa** [reply]
What I get is the following:

![image](https://github.com/rangav/thunder-client-support/assets/40989280/760de560-0fdd-4698-a171-ae6f1c93b56e)

**Reply by rangav** [reply]
The request url host should match the host in the thunder client certificates setting

so please use `hpims.com` instead of `localhost:8081` as below

```
"thunder-client.certificates": [
        {
            "host": "hpims.com",
            "pfxPath": "C:/Users/terrami/Documents/HP-IMS Related Documents/SSL Certificates/MyCert.pfx",
            "passphrase": "xxxx"
        },
    ]
```

**Reply by mith-emsa** [reply]
Hi @rangav I changed to emsa.europa.eu , because that is the host of my request url. But I still got the error, but now with some useful output in the terminal. See below
![image](https://github.com/rangav/thunder-client-support/assets/40989280/841513e7-7da9-439f-a07f-e97ccf29fe3c)

**Reply by mith-emsa** [reply, marked as answer]
Hi @rangav I changed the path to 
`"C:\\Users\\terrami\\Documents\\HP-IMS Related Documents\\SSL Certificates\\MyCert.pfx" `
and put the host to **emsa.europa.eu** and now it works.

**Reply by rangav** [reply]
Super, finally you got it working
